values_test.go

```diff
@@ -0,0 +1,198 @@
+			NewInt256(-1):                 {255},
+			NewInt256(-200):               {255, 56},
+			NewInt256(-10000000000000000): {220, 121, 13, 144, 63, 0, 0},
+		},
+		// UInt*
+		"UInt": {
+			NewUInt(0):   {0},
+			NewUInt(42):  {42},
+			NewUInt(127): {127},
+			NewUInt(128): {128},
+			NewUInt(200): {200},
+		},
+		"UInt8": {
+			NewUInt8(0):   {0},
+			NewUInt8(42):  {42},
+			NewUInt8(127): {127},
+			NewUInt8(128): {128},
+			NewUInt8(255): {255},
+		},
+		"UInt16": {
+			NewUInt16(0):     {0, 0},
+			NewUInt16(42):    {0, 42},
+			NewUInt16(32767): {127, 255},
+			NewUInt16(32768): {128, 0},
+			NewUInt16(65535): {255, 255},
+		},
+		"UInt32": {
+			NewUInt32(0):          {0, 0, 0, 0},
+			NewUInt32(42):         {0, 0, 0, 42},
+			NewUInt32(2147483647): {127, 255, 255, 255},
+			NewUInt32(2147483648): {128, 0, 0, 0},
+			NewUInt32(4294967295): {255, 255, 255, 255},
+		},
+		"UInt64": {
+			NewUInt64(0):                    {0, 0, 0, 0, 0, 0, 0, 0},
+			NewUInt64(42):                   {0, 0, 0, 0, 0, 0, 0, 42},
+			NewUInt64(9223372036854775807):  {127, 255, 255, 255, 255, 255, 255, 255},
+			NewUInt64(9223372036854775808):  {128, 0, 0, 0, 0, 0, 0, 0},
+			NewUInt64(18446744073709551615): {255, 255, 255, 255, 255, 255, 255, 255},
+		},
+		"UInt128": {
+			NewUInt128(0):   {0},
+			NewUInt128(42):  {42},
+			NewUInt128(127): {127},
+			NewUInt128(128): {128},
+			NewUInt128(200): {200},
+		},
+		"UInt256": {
+			NewUInt256(0):   {0},
+			NewUInt256(42):  {42},
+			NewUInt256(127): {127},
+			NewUInt256(128): {128},
+			NewUInt256(200): {200},
+		},
+		// Word*
+		"Word8": {
+			NewWord8(0):   {0},
+			NewWord8(42):  {42},
+			NewWord8(127): {127},
+			NewWord8(128): {128},
+			NewWord8(255): {255},
+		},
+		"Word16": {
+			NewWord16(0):     {0, 0},
+			NewWord16(42):    {0, 42},
+			NewWord16(32767): {127, 255},
+			NewWord16(32768): {128, 0},
+			NewWord16(65535): {255, 255},
+		},
+		"Word32": {
+			NewWord32(0):          {0, 0, 0, 0},
+			NewWord32(42):         {0, 0, 0, 42},
+			NewWord32(2147483647): {127, 255, 255, 255},
+			NewWord32(2147483648): {128, 0, 0, 0},
+			NewWord32(4294967295): {255, 255, 255, 255},
+		},
+		"Word64": {
+			NewWord64(0):                    {0, 0, 0, 0, 0, 0, 0, 0},
+			NewWord64(42):                   {0, 0, 0, 0, 0, 0, 0, 42},
+			NewWord64(9223372036854775807):  {127, 255, 255, 255, 255, 255, 255, 255},
+			NewWord64(9223372036854775808):  {128, 0, 0, 0, 0, 0, 0, 0},
+			NewWord64(18446744073709551615): {255, 255, 255, 255, 255, 255, 255, 255},
+		},
+		//// Fix*
+		//"Fix64": {
+		//	NewFix64("0.0"):   {0, 0, 0, 0, 0, 0, 0, 0},
+		//	NewFix64("42.0"):  {0, 0, 0, 0, 250, 86, 234, 0},
+		//	NewFix64("42.24"): {0, 0, 0, 0, 251, 197, 32, 0},
+		//	NewFix64("-1.0"):  {255, 255, 255, 255, 250, 10, 31, 0},
+		//},
+		//// UFix*
+		//"UFix64": {
+		//	NewFix64("0.0"):   {0, 0, 0, 0, 0, 0, 0, 0},
+		//	NewFix64("42.0"):  {0, 0, 0, 0, 250, 86, 234, 0},
+		//	NewFix64("42.24"): {0, 0, 0, 0, 251, 197, 32, 0},
+		//},
```

Comment:
Are you saving these for a separate PR?

Reply:
No, this was a work in progress, the PR should be ready as of 9953a94